src/generated/graphql.tsx

```diff
@@ -2267,356 +1365,260 @@ export enum _SubgraphErrorPolicy_ {
-    totalSupply
+    ohmTotalSupply
+    sOhmCirculatingSupply
+    timestamp
     totalValueLocked
-    treasuryLiquidBackingPerGOhm
-    treasuryLiquidBackingPerOhmFloating
-    treasuryMarketValue
   }
 }
     `;
-export const useMetricsBarLatestOnlyQuery = <TData = MetricsBarLatestOnlyQuery, TError = unknown>(
+export const useProtocolMetricsQuery = <TData = ProtocolMetricsQuery, TError = unknown>(
   dataSource: { endpoint: string; fetchParams?: RequestInit },
-  variables?: MetricsBarLatestOnlyQueryVariables,
-  options?: UseQueryOptions<MetricsBarLatestOnlyQuery, TError, TData>,
+  variables: ProtocolMetricsQueryVariables,
+  options?: UseQueryOptions<ProtocolMetricsQuery, TError, TData>,
 ) =>
-  useQuery<MetricsBarLatestOnlyQuery, TError, TData>(
-    variables === undefined ? ["MetricsBarLatestOnly"] : ["MetricsBarLatestOnly", variables],
-    fetcher<MetricsBarLatestOnlyQuery, MetricsBarLatestOnlyQueryVariables>(
+  useQuery<ProtocolMetricsQuery, TError, TData>(
+    ["ProtocolMetrics", variables],
+    fetcher<ProtocolMetricsQuery, ProtocolMetricsQueryVariables>(
       dataSource.endpoint,
       dataSource.fetchParams || {},
-      MetricsBarLatestOnlyDocument,
+      ProtocolMetricsDocument,
       variables,
     ),
     options,
   );
-export const KeyMetricsDocument = `
-    query KeyMetrics($records: Int = 100) {
-  protocolMetrics(first: $records, orderBy: timestamp, orderDirection: desc) {
-    id
-    block
-    currentIndex
-    gOhmPrice
-    gOhmTotalSupply
-    marketCap
-    ohmCirculatingSupply
-    ohmFloatingSupply
-    ohmPrice
-    timestamp
-    timestampISO8901
-    totalSupply
-    totalValueLocked
-    treasuryLiquidBacking
-    treasuryLiquidBackingPerGOhm
-    treasuryLiquidBackingPerOhmFloating
-    treasuryMarketValue
-  }
-}
-    `;
-export const useKeyMetricsQuery = <TData = KeyMetricsQuery, TError = unknown>(
+export const useInfiniteProtocolMetricsQuery = <TData = ProtocolMetricsQuery, TError = unknown>(
   dataSource: { endpoint: string; fetchParams?: RequestInit },
-  variables?: KeyMetricsQueryVariables,
-  options?: UseQueryOptions<KeyMetricsQuery, TError, TData>,
+  _pageParamKey: keyof ProtocolMetricsQueryVariables,
+  variables: ProtocolMetricsQueryVariables,
+  options?: UseInfiniteQueryOptions<ProtocolMetricsQuery, TError, TData>,
 ) =>
-  useQuery<KeyMetricsQuery, TError, TData>(
-    variables === undefined ? ["KeyMetrics"] : ["KeyMetrics", variables],
-    fetcher<KeyMetricsQuery, KeyMetricsQueryVariables>(
-      dataSource.endpoint,
-      dataSource.fetchParams || {},
-      KeyMetricsDocument,
-      variables,
-    ),
+  useInfiniteQuery<ProtocolMetricsQuery, TError, TData>(
+    ["ProtocolMetrics.infinite", variables],
+    metaData =>
+      fetcher<ProtocolMetricsQuery, ProtocolMetricsQueryVariables>(
+        dataSource.endpoint,
+        dataSource.fetchParams || {},
+        ProtocolMetricsDocument,
+        { ...variables, ...(metaData.pageParam ?? {}) },
+      )(),
```

Comment:
this looks potentially problematic. There's a potential that your endpoint, fetchParams, and page params will change, however it looks like your query key is only tracking 'variables' , which doesn't seem to include those items.

Reply:
this feedback applies to all the generated queries btw. I think this is potentially the root of some of the data refresh issues.

Reply:
We're not using multiple endpoints, so I don't think that's an issue.

The base variables (e.g. limit category to POL) won't change, but the pageParam (date <= X and date > Y) will. Do you think those should be distinct query keys?

The query issue did take place when the treasury assets chart was displaying market value and then switched to liquid backing. The treasury assets table would observe the change in the variable representing that, and then trigger a refetch of the data. That ended up with one of the variables changing (https://github.com/OlympusDAO/olympus-frontend/blob/f0df15171aff42aa642f32ee17f0a9a78ba89b0d/src/views/TreasuryDashboard/components/Graph/TreasuryAssetsTable.tsx#L47) within a `useMemo` block, so `useInfiniteTokenRecordsQuery` should have used the updated variable in its key. It's very weird.

Reply:
> The base variables (e.g. limit category to POL) won't change, but the pageParam (date <= X and date > Y) will. Do you think those should be distinct query keys?

If these are being used inside the query itself then they should be added to the key. You're probably OK since pageParam and date ranges in the variables object are likely changing at the same time, but it is a best practice to include any variable that can change in the query as part of the key: 
https://tanstack.com/query/v4/docs/guides/query-keys#if-your-query-function-depends-on-a-variable-include-it-in-your-query-key

Reply:
I think what's happening is that the _initial_ variables (e.g. date > 1st August 2022 and < 1st September 2022) are included in the key, but the subsequent page params (date > 1st July, date > 1st June, etc) are not in the query key. I'll follow that thread when trying to isolate the problem and file a bug report. Thanks!